reaktive/src/commonMain/kotlin/com/badoo/reaktive/observable/Amb.kt

```diff
@@ -0,0 +1,59 @@
+package com.badoo.reaktive.observable
+
+import com.badoo.reaktive.disposable.CompositeDisposable
+import com.badoo.reaktive.disposable.Disposable
+import com.badoo.reaktive.disposable.minusAssign
+import com.badoo.reaktive.disposable.plusAssign
+import com.badoo.reaktive.utils.ObjectReference
+
+fun <T> Iterable<Observable<T>>.amb(): Observable<T> =
+    observable { emitter ->
+        val disposables = CompositeDisposable()
+        emitter.setDisposable(disposables)
+        val markerDisposable = Disposable()
```

Comment:
Why do we need `markerDisposable`? Simple `AtomicBoolean` won't work?

Reply:
Good catch, fixed